### PR TITLE
handle incoming event sequentially by limiting dispatcher

### PR DESCRIPTION
**Pull request by rcmkt**

### CODE REVIEW

File: elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt [hunk 1]
```diff
@@ -62,37 +65,30 @@ class ElmStore<Event : Any, State : Any, Effect : Any, Command : Any>(
...
-            } catch (error: CancellationException) {
-                throw error
-            } catch (t: Throwable) {
-                storeListeners.forEach { it.onReducerError(key, t, event) }
-                logger.fatal(
-                    message = "You must handle all errors inside reducer",
-                    tag = key,
-                    error = t,
-                )
+    private fun CoroutineScope.handleEvent(event: Event) = launch(eventDispatcher) {
```
Comment (opengamer29) [inline elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt:68]: context(CoroutineScope)?

File: elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt [hunk 2]
```diff
@@ -13,27 +15,26 @@ import kotlinx.coroutines.flow.catch
...
-        (reducer::class.qualifiedName?: reducer::class.simpleName).orEmpty().replace(
+        (reducer::class.qualifiedName ?: reducer::class.simpleName).orEmpty().replace(
             "Reducer",
             "Store",
         ),
 ) : Store<Event, Effect, State> {
 
     private val logger = ElmslieConfig.logger
-    private val eventMutex = Mutex()
+    private val eventDispatcher = ElmslieConfig.ioDispatchers.limitedParallelism(parallelism = 1)
```
Comment (opengamer29) [inline elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt:37]: to guarantee a sequence of events, it is better to use coroutines channels

Comment (rcmkt) [reply elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt:37]: =) not really

File: elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt (hunk 1 above)
Comment (rcmkt) [reply elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt:68]: ?

File: elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (rcmkt) [reply elmslie-core/src/commonMain/kotlin/money/vivid/elmslie/core/store/ElmStore.kt:37]: i don't think its good idea, this change is related only to reduce event. Reducing really should be sequential always, it's not good when some event1 was dispatched before event2 but handled the last.
After reducing, executing commands and sending effect has the same behaviour

**Review by sergei-lapin — APPROVED** [review]

**Review by opengamer29 — APPROVED** [review]

**Pull request merged**